src/HOCON/Impl/Utils.cs

```diff
@@ -0,0 +1,183 @@
+            int index = 0;
+            while (Whitespaces.Contains(value[index]))
+            {
+                index++;
+            }
+
+            var trimmed = value.Substring(index);
+            index = trimmed.Length - 1;
+            while (Whitespaces.Contains(value[index]))
+            {
+                index--;
+            }
+            var res = trimmed.Substring(0, index + 1);
+            return res;
+        }
+
+        public static bool IsHoconWhitespace(this char c)
+            => Whitespaces.Contains(c);
+
+        public static bool IsWhitespaceWithNoNewLine(this char c)
+            => WhitespaceWithoutNewLine.Contains(c);
+
+        public static bool IsDigit(this char c)
+            => Digits.Contains(c);
+
+        public static bool IsOctal(this char c)
+            => Octals.Contains(c);
+
+        public static bool IsHexadecimal(this char c)
+            => Hexadecimal.Contains(c);
+
+        public static string ToHoconSafe(this string s)
+            => s.NeedTripleQuotes() ? $"\"\"\"{s}\"\"\"" : s.NeedQuotes() ? $"\"{s}\"" : s;
+
+        public static HoconPath ToHoconPath(this string path)
+            => HoconPath.Parse(path);
+
+        #endregion
+
+        #region Token extensions
+
+        [DebuggerStepThrough]
+        public static bool IsSignificant(this Token token)
+            => token.Type != TokenType.Comment
+               && token.Type != TokenType.EndOfLine
+               && token.LiteralType != TokenLiteralType.Whitespace;
+
+        [DebuggerStepThrough]
+        public static bool IsNonSignificant(this Token token)
+            => !token.IsSignificant();
+
+        #endregion
+
+        public static bool IsSubstitution(this IHoconElement value)
+        {
+            switch (value)
+            {
+                case HoconValue v:
+                    foreach (var val in v)
+                    {
+                        if (val is HoconSubstitution)
+                            return true;
+                    }
+                    return false;
+                case HoconField f:
+                    return f.Value.OfType<HoconSubstitution>().Any();
+                case HoconSubstitution _:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+    }
+
+    public static class StringUtil
+    {
+
+        public static bool NeedQuotes(this string s)
+            => s.Intersect(Utils.NotInUnquotedText).Any();
+
+        public static bool NeedTripleQuotes(this string s)
+            => s.NeedQuotes() && s.Contains(Utils.NewLine);
+
+    }
+
+    internal static class AsyncHelper
+    {
+        private static readonly TaskFactory TaskFactory = new
+            TaskFactory(CancellationToken.None,
+                TaskCreationOptions.None,
+                TaskContinuationOptions.None,
+                TaskScheduler.Default);
+
+        public static TResult RunSync<TResult>(Func<Task<TResult>> func)
```

Comment:
How do these functions differ from `func().GetAwaiter().GetResult()`?

Reply:
To be honest, I don't know. I lifted the code from 
https://github.com/aspnet/AspNetIdentity/blob/master/src/Microsoft.AspNet.Identity.Core/AsyncHelper.cs
The biggest concern is that `func().GetAwaiter().GetResult()` can cause a deadlock related to context. I kind of figured that Microsoft knows their own code better than I do, so I tried to code as defensively as possible.